examples/src/main/java/com/mapbox/navigation/examples/core/MapboxDropInActivity.kt

```diff
@@ -31,5 +43,58 @@ class MapboxDropInActivity : AppCompatActivity() {
         binding.tempStartNavigation.setOnClickListener {
             mapboxNavigationViewApi.temporaryStartNavigation()
         }
+
+        mapboxNavigationViewApi.getMapView()
+            .gestures
+            .addOnMapLongClickListener(onMapLongClickListener)
```

Comment:
```suggestion
        MapLongClickRouteRequest().observeClicks(binding.mapView, lifecycle)
```

Can we make a drop-in-ui component to do all this? Seems wrong to have the MapboxDropInActivity include `WeakReference` and a bunch of boilerplate. MapLongClickRouteRequest is something we can encapsulate

Reply:
I don't think it's intended that the drop in UI support or implement a map long click.  It was put in as a temporary measure to get a route into mapbox navigation so that development could continue.  At that time there was no API support for getting a route into mapbox navigation via the `NavigationView`.  Since this work addresses that API functionality the (temporary) long click was moved out of `NavigationView`. However, in order to continue development we still need some way of getting a route into `NavigationView` which is why the long press was moved into the example activity.  Each developer will have their own way of getting a route into `NavigationView` whether it be search and/or map long press and/or other.  The example activity is using long press so we have an easy way of continuing  development on the drop in UI.

Reply:
But it's also not intended to have a big `MapboxDropInActivity`. We're trying to create a proof that our navigation is easy to integrate, so there should be some way to simplify this. 

What do you think about the idea in the [AppLifecycleActivity](https://github.com/mapbox/mapbox-navigation-android/blob/3751b1323b3e585976a5476176810049061ca1ec/qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/AppLifecycleActivity.kt#L96-L101)?

That could be simplified too. Can we play with ideas in the `qa-test-app`? I wouldn't want to recommend WeakReferences to use our drop-in-ui. We understand them, but they are still tricky. The example `MapboxDropInActivity` should showcase our recommendation for a simple integration

Reply:
The activity has been reverted to its previous state.